tcms/xmlrpc/serializer.py

```diff
@@ -237,14 +289,14 @@ def _get_primary_key_field(self):
         """
         if hasattr(self.__class__, 'primary_key'):
             return self.__class__.primary_key
-        else:
-            fields = [field.name for field in self.model_class._meta.fields
-                      if field.primary_key]
-            if not fields:
-                raise ValueError(
-                    'Model %s has no primary key. You have to specify such '
-                    'field manually.' % self.model_class.__name__)
-            return fields[0]
+
+        for field in self.model_class._meta.fields:
+            if field.primary_key:
+                return field.name
+
+        raise ValueError(
+            _('Model %s has no primary key. You have to specify such '
+              'field manually.' % self.model_class.__name__))
```

Comment:
The `_()` function must accept the string, can be a format string and it will return a string. Then the result will apply `% self.model_class.__name__` on the translation. The actual model name should not be part of the translation string.